### PR TITLE
OboeTester: Fix input channel masks data path tests

### LINKED ISSUE #1869: OboeTester Data path tests don't look at input channel masks correctly
Should reference [AAudioConvert_androidToAAudioChannelLayoutMask](https://cs.android.com/android/platform/superproject/main/+/refs/heads/main:frameworks/av/media/libaaudio/src/utility/AAudioUtilities.cpp;l=367;drc=3d4951806618b5119ec60a12d80c026bf7ae4153)

### PR DESCRIPTION
**Pull request by robertwu1**
Fixes #1869

### CODE REVIEW

File: apps/OboeTester/app/src/main/java/com/mobileer/oboetester/TestDataPathsActivity.java [hunk 1]
```diff
@@ -70,6 +70,35 @@ public class TestDataPathsActivity  extends BaseAutoGlitchActivity {
...
+    public static final int JAVA_CHANNEL_IN_BACK_RIGHT = 1 << 17;
+    public static final int JAVA_CHANNEL_IN_CENTER = 1 << 18;
+    public static final int JAVA_CHANNEL_IN_LOW_FREQUENCY = 1 << 20;
+    public static final int JAVA_CHANNEL_IN_TOP_LEFT = 1 << 21;
+    public static final int JAVA_CHANNEL_IN_TOP_RIGHT = 1 << 22;
+
+    public static final int JAVA_CHANNEL_IN_MONO = JAVA_CHANNEL_IN_FRONT;
+    public static final int JAVA_CHANNEL_IN_STEREO = JAVA_CHANNEL_IN_LEFT | JAVA_CHANNEL_IN_RIGHT;
+    public static final int JAVA_CHANNEL_IN_FRONT_BACK = JAVA_CHANNEL_IN_FRONT | JAVA_CHANNEL_IN_BACK;
+    public static final int JAVA_CHANNEL_2POINT0POINT2 = JAVA_CHANNEL_IN_LEFT |
```
Comment (philburk) [inline apps/OboeTester/app/src/main/java/com/mobileer/oboetester/TestDataPathsActivity.java:87]: Should this be JAVA_CHANNEL_IN_2POINT0POINT2 ?

File: apps/OboeTester/app/src/main/java/com/mobileer/oboetester/TestDataPathsActivity.java [hunk 2]
```diff
@@ -70,6 +70,35 @@ public class TestDataPathsActivity  extends BaseAutoGlitchActivity {
...
+            JAVA_CHANNEL_IN_RIGHT |
+            JAVA_CHANNEL_IN_TOP_LEFT |
+            JAVA_CHANNEL_IN_TOP_RIGHT;
+    public static final int JAVA_CHANNEL_2POINT1POINT2 = JAVA_CHANNEL_2POINT0POINT2 | JAVA_CHANNEL_IN_LOW_FREQUENCY;
+    public static final int JAVA_CHANNEL_3POINT0POINT2 = JAVA_CHANNEL_2POINT0POINT2 | JAVA_CHANNEL_IN_CENTER;
+    public static final int JAVA_CHANNEL_3POINT1POINT2 = JAVA_CHANNEL_3POINT0POINT2 | JAVA_CHANNEL_IN_LOW_FREQUENCY;
+    public static final int JAVA_CHANNEL_5POINT1 = JAVA_CHANNEL_IN_LEFT |
```
Comment (philburk) [inline apps/OboeTester/app/src/main/java/com/mobileer/oboetester/TestDataPathsActivity.java:94]: Ditto, all these could be _IN_

**Review by philburk — COMMENTED** [review]
With this change, is DataPaths able to open all the Mic DeviceInfo masks?

File: apps/OboeTester/app/src/main/java/com/mobileer/oboetester/TestDataPathsActivity.java [hunk 3]
```diff
@@ -70,6 +70,35 @@ public class TestDataPathsActivity  extends BaseAutoGlitchActivity {
     private final static double MAX_ALLOWED_JITTER = 2.0 * PHASE_PER_BIN;
     private final static String MAGNITUDE_FORMAT = "%7.5f";
 
+    public static final int JAVA_CHANNEL_IN_LEFT = 1 << 2;
```
[the remark this answers is not in the document]
Comment (philburk) [reply apps/OboeTester/app/src/main/java/com/mobileer/oboetester/TestDataPathsActivity.java:73]: StreamConfiguration says 
CHANNEL_FRONT_LEFT = 1 << 0;
CHANNEL_FRONT_CENTER = 1 << 2;

So the mappings coming from Java DeviceInfo do not align.

[the remark this answers is not in the document]
Comment (robertwu1) [reply apps/OboeTester/app/src/main/java/com/mobileer/oboetester/TestDataPathsActivity.java:73]: For the output case, we can simply shift by 2 bits. Input is just different with AAudioConvert_aaudioToAndroidChannelLayoutMask() mapping input to something like output

**Review by philburk — APPROVED** [review]

**Pull request merged**